refresh.template.py

```diff
@@ -767,6 +745,73 @@ def _apple_platform_patch(compile_args: typing.List[str]):
     return compile_args
 
 
+def _get_sysroot(args: typing.List[str]):
+    for idx, arg in enumerate(args):
+        if arg == '--sysroot' or arg == '-isysroot':
+            if idx + 1 < len(args):
+                return pathlib.PurePath(args[idx + 1])
+        elif arg.startswith('--sysroot='):
+            return pathlib.PurePath(arg[len('--sysroot='):])
+        elif arg.startswith('-isysroot'):
+            return pathlib.PurePath(arg[len('-isysroot'):])
+    return None
+
+
+def _emscripten_platform_patch(compile_args: typing.List[str]):
+    """De-Bazel the command into something clangd can parse.
+
+    This function has fixes specific to Emscripten platforms, but you should call it on all platforms. It'll determine whether the fixes should be applied or not
+    """
+    emcc_driver = pathlib.Path(compile_args[0])
+    if not emcc_driver.name.startswith('emcc'):
+        return compile_args
+
+    workspace_absolute = pathlib.PurePath(os.environ["BUILD_WORKSPACE_DIRECTORY"])
+    sysroot = _get_sysroot(compile_args)
+    assert sysroot, f'Emscripten sysroot not detected in CMD: {compile_args}'
+
+    def get_workspace_root(path_from_execroot: pathlib.PurePath):
+        assert path_from_execroot.parts[0] == 'external'
```

Comment:
*super* minor, but I suppose this might not be external if invoked within the emsdk...would we want to return `.` rather than assert?

Reply:
Sure, I fixed it.